code/modules/shuttle/computer.dm

```diff
@@ -366,12 +366,42 @@
 	breakable = FALSE
 	///If true, the lifeboat is in the process of launching, and so the code will not allow another launch.
 	var/launch_initiated = FALSE
+	///If true, the lifeboat is in the process of having the xeno override removed by the pilot.
+	var/override_being_removed = FALSE
 
 /obj/structure/machinery/computer/shuttle/lifeboat/attack_hand(mob/user)
 	. = ..()
 	var/obj/docking_port/mobile/crashable/lifeboat/lifeboat = SSshuttle.getShuttle(shuttleId)
 	if(lifeboat.status == LIFEBOAT_LOCKED)
-		to_chat(user, SPAN_WARNING("[src] flickers with error messages."))
+		if(skillcheck(user, SKILL_PILOT, SKILL_PILOT_TRAINED))
+			if(user.action_busy)
+				return
+			else if(!override_being_removed)
+				to_chat(user, SPAN_NOTICE("You start to remove the lockout."))
+				override_being_removed = TRUE
+				var/unhack_time = 1800
+				user.visible_message(SPAN_NOTICE("[user] starts to type on [src]."),
+					SPAN_NOTICE("You try to take back control over the lifeboat. It will take around 3 minutes."))
+				var/remaining_time = timeleft(unhack_time) / 10
+				while(remaining_time > 20)
+					if(!do_after(user, 20 SECONDS, INTERRUPT_ALL|INTERRUPT_CHANGED_LYING, BUSY_ICON_HOSTILE, numticks = 20))
+						to_chat(user, SPAN_WARNING("You fail to remove the lockout!"))
+						override_being_removed = FALSE
+						return
+					remaining_time = timeleft(unhack_time) / 10 - 20
```

Comment:
```suggestion
					remaining_time = remaining_time - 20 SECONDS 
```

What you coded definitely doesn't work as it should, it doesn't save your progress if you get interrupted. Just make var/remaining_time = 180 SECONDS in the object, make sure it resets to initial on repeated locking, and just subtract 20 SECONDS every cycle.

Reply:
I suggested everything except resetting the timer on repeated locking, find where Queen locks lifeboat and put ```remaining_time = initial(remaining_time)```
there